docs/tasks/manage-stateful-set/upgrade-pet-set-to-stateful-set.md

```diff
@@ -0,0 +1,112 @@
+---
+assignees:
+- bprashanth
+- enisoc
+- erictune
+- foxish
+- janetkuo
+- kow3ns
+- smarterclayton
+
+---
+
+{% capture overview %}
+This page shows how to upgrade from Pet Sets (Kubernetes cluster at version 1.3-1.4) to Stateful Sets (version >= 1.5).
+{% endcapture %}
+
+{% capture prerequisites %}
+
+* If you don't have Pet Sets in your current cluster, or you don't plan to upgrade your cluster/master to >= 1.5 at this moment, you can skip this task now. 
+
+{% endcapture %}
+
+{% capture steps %}
+
+### Differences between alpha Pet Sets and beta Stateful Sets
+
+Pet Set was introduced as an alpha resource in Kubernetes release 1.3, and was renamed to Stateful Set as a beta resource in 1.5. 
+Here are some notable changes:
+
+* **Stateful Set is the new Pet Set**: Pet Set is no longer available in any Kubernetes release >= 1.5. It becomes beta Stateful Set. To know why the name is changed, see this [discussion thread](https://github.com/kubernetes/kubernetes/issues/27430).
+* **Stateful Set guards against split brain**: Stateful Sets guarantee at most one pod for a given ordinal index can be running anywhere in a cluster, in order to guard against split brain scenarios with distributed applications. *TODO: Link to doc about fencing*
+* **Flipped debug annotation behavior**: The default value of the debug annotation (`pod.alpha.kubernetes.io/initialized`) is now `true`. The absence of this annotation will pause Pet Set operations, but will NOT pause Stateful Set operations. In most cases, you no longer need this annotation in your Stateful Set manifests. 
+
+
+### Upgrading from Pet Sets to Stateful Sets
+
+#### Find all Pet Sets and their manifests 
```

Comment:
I was worried that if the user upgrades to kubectl from v1.4 to v1.5 first, then the following (`kubectl get petsets`) would not work. I tested it just now and it does seem to work, but there may be some unexpected results. For example, I did `kubectl delete petset <name>` with kubectl v1.5 and it did delete the petset, but it didn't scale it down first. I'm guessing that's because the scale-down is initiated by kubectl on the client side, so if you delete a type that the client doesn't know about, it will be the same as `--cascade=false`?